src/scales/scale.time.js

```diff
@@ -28,6 +32,18 @@ INTERVALS.set('month', {common: true, size: 2.628e9, steps: 12});
 INTERVALS.set('quarter', {common: false, size: 7.884e9, steps: 4});
 INTERVALS.set('year', {common: true, size: 3.154e10, steps: undefined});
 
+/**
+ * @param {Unit} unit
+ * @return {Interval}
+ */
+function getInterval(unit) {
```

Comment:
Using `Map` made me think its probably slower in this use case. Adding a function that can throw probably makes it quite a lot slower. Just a hunch.

Reply:
This function isn't called that many times, so I doubt it's noticable

Reply:
Whats the gain of having a `Map` though? 
If it just for the `@type`, I think you could use `{Object<Unit, Interval>}` too. No strong opinions either way though, just resisting a change that does not have a purpose. (that was already merged, so a bit late here)

Reply:
The issue is it's really hard to build the `UNITS` array from the keys of an object because `Object.keys` returns `string[]` instead of `Unit[]` (https://github.com/microsoft/TypeScript/pull/12253#issuecomment-263132208). Supposedly I `for ... in` is supposed to work, but it didn't for me. I'm happy to go back to a map if you can get it to work, but I couldn't figure it out after spending a decent amount of time on it and am not sure it's worth more effort. Most of the solutions I find online are for TypeScript code and can't really be applied in JavaScript code, so I just gave up and used a `Map`

Reply:
https://github.com/chartjs/Chart.js/blob/decb57b444cb5a6652f938d0426f42c3cbe872ab/src/scales/scale.time.js#L61

Reply:
These are the types of things I don't like typescript for. It makes things hard while not providing any runtime type safety. Its good up to a point, but then it just gets silly.

There are quite a few good things in this PR, but I don't think we should merge all of these changes.

Reply:
Ah, thanks for the typecast hint. I switched this back to an object